start.js

```diff
@@ -3,3 +3,13 @@ const app = require('./server')
 app.listen(app.port, function () {
   console.error(`${app.name} listening on port ${app.port} [${app.env}]`)
 })
+
+
+process.on('SIGINT', function(){
```

Comment:
For a process to be run as PID 1 via Docker, the signal handlers are required otherwise a PID 1 process [won't take action on a signal](https://github.com/docker/docker/pull/3240#issuecomment-31035598). With the `Dockerfile` change you need the signal handlers. 

It could use [130 (128+2) on a `SIGINT`](http://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF).  I would normally expect a `SIGTERM` to be standard shutdown and return 0 though, unless there were issues in the handler code. It seems 143 can be common (128+15) 

The handlers are also useful to house logic that does something like wait for any current downloads to finish/timeout or clean up any currently writing files. I can definitely cause the truncation in https://github.com/dickeyxxx/npm-register/issues/48 by terminating an actively writing server.